modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,220 @@
+    if (!this.config.accountSID || !this.config.authToken) {
+      return this.logger.error(`[${this.botId}] The accountSID and authToken must be configured to use this channel.`)
+    }
+
+    const url = (await this.router.getPublicPath()) + this.route
+    this.webhookUrl = url.replace('BOT_ID', this.botId)
+
+    this.twilio = new Twilio(this.config.accountSID, this.config.authToken)
+    this.kvs = this.bp.kvs.forBot(this.botId)
+
+    this.logger.info(`Twilio webhook listening at ${this.webhookUrl}`)
+  }
+
+  auth(req): boolean {
+    const signature = req.headers['x-twilio-signature']
+    return validateRequest(this.config.authToken, signature, this.webhookUrl, req.body)
+  }
+
+  getKvsKey(target: string, threadId: string) {
+    return `${target}_${threadId}`
+  }
+
+  async handleWebhookRequest(body: TwilioRequestBody) {
+    debugIncoming(`Received message %o`, body)
+
+    const threadId = body.To
+    const target = body.From
+    const text = body.Body
+
+    const index = Number(text)
+    let payload: any = { type: 'text', text }
+    if (index) {
+      payload = (await this.handleIndexReponse(index - 1, target, threadId)) ?? payload
+      if (payload.type === 'url') {
+        return
+      }
+    }
+
+    await this.kvs.delete(this.getKvsKey(target, threadId))
+
+    await this.bp.events.sendEvent(
+      this.bp.IO.Event({
+        botId: this.botId,
+        channel: 'twilio',
+        direction: 'incoming',
+        type: payload.type,
+        payload,
+        threadId,
+        target
+      })
+    )
+  }
+
+  async handleIndexReponse(index: number, target: string, threadId: string): Promise<any> {
+    const key = this.getKvsKey(target, threadId)
+    if (!(await this.kvs.exists(key))) {
+      return
+    }
+
+    const option = await this.kvs.get(key, `[${index}]`)
+    if (!option) {
+      return
+    }
+
+    await this.kvs.delete(key)
+
+    const { type, label, value } = option
+    return {
+      type,
+      text: type === 'say_something' ? value : label,
+      payload: value
+    }
+  }
+
+  async handleOutgoingEvent(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    const { type, payload } = event
+    const { __buttons, __dropdown } = payload.metadata as sdk.Content.Metadata
+
+    if (__buttons) {
+      await this.sendChoices(event)
+    } else if (__dropdown) {
+      await this.sendDropdown(event)
+    } else if (type === 'text') {
+      await this.sendText(event)
+    } else if (type === 'image') {
+      await this.sendImage(event)
+    } else if (type === 'carousel') {
+      await this.sendCarousel(event)
+    }
+    next(undefined, false)
+  }
+
+  async sendText(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.text
+    })
+  }
+
+  async sendImage(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.title,
+      mediaUrl: [`${event.payload.BOT_URL}${event.payload.image}`]
+    })
+  }
```

Comment:
Image is not working. If you use the correct typings, you would see that BOT_URL is not at the correct place

Reply:
I changed it so it matches the typings altough BOT_URL was present at bot places for some reason